client-tags/delivered-and-read.md

```diff
@@ -0,0 +1,73 @@
+---
+title: IRCv3 delivered and read client tags
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "evan@muffinmedic.net"
+---
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+delivered` and `+read` tag names. Instead, implementations SHOULD use the
+`+draft/delivered` and `+draft/read` tag names to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use unprefixed tag names.
+
+## Description
+The specification includes two client tags for message tracking: `delivered` and `read`. These tags provide for delivery and read receipts, respectively, and enable users to know when and by whom a message has been delivered to or read.
+
+For the purpose of this specification, the terms "delivery receipt" and "read receipt" are synonymous with the use of the `delivered` and `read` client tags, respectively.
+
+For the purpose of this specification, a message is defined as a user-generated `PRIVMSG`, `NOTICE`, `TAGMSG`, or [`batch`][batch] end.
+
+## Format
+`delivered` and `read` are sent as a [`TAGMSG`][tags] and must include the [`draft/msgid`][id] of the original message as the client tag parameter.
+
+Both client tags follow the same format. Clients MUST send a `delivered` tag immediately upon receipt of the message, including the [`draft/msgid`][id] as the client tag parameter. Clients MUST send a `read` tag when the message is first brought into view (e.g. the correct target with the corresponding message is in focus). If the message is delivered and read at the same time, clients MAY send both tags in the same `TAGMSG` or send `delivered` followed by `read` as separate `TAGMSG`s. Clients SHOULD NOT elect to send only a `read` tag and MUST NOT send `read` before `delivered`.
+
+A `delivered` and `read` tag MUST be sent for each unique `draft/msgid`. If a [`batch`][batch] is received, the client MUST NOT send a single `delivered` and `read` for the entirety of the batch. Clients MAY send `delivered` and `read` tags before a [`batch`][batch] end is received. Upon receipt of a `delivered` or `read` tag, clients SHOULD assume all previous messages in the given target have also been delivered or read by the user as appropriate.
```

Comment:
Yeah in general it would be nice to have all the sending features opt-in, and being able to be compliant with both specs by supporting receiving.